server/routes/api/browser.py

```diff
@@ -164,10 +163,49 @@ def get_statvar_hierarchy(dcid):
     for svg_id, svg in svg_map.items():
         if svg_id in seen_sv:
             continue
+        svg["level"] = 0
         processed_svg_map[svg_id] = svg
         statvar_hierarchy_helper(svg_id, svg_map, processed_svg_map,
-                                 processed_sv, seen_sv)
+                                 processed_sv, seen_sv, 1)
+    for sv_id, sv in processed_sv.items():
+        parent_svg = processed_svg_map.get(sv["parent"])
+        sv["level"] = parent_svg["level"] + 1
     result = {}
     result["statVarGroups"] = processed_svg_map
     result["statVars"] = processed_sv
     return Response(json.dumps(result), 200, mimetype='application/json')
+
+
+@cache.memoize(timeout=3600 * 24)  # Cache for one day.
+def search_helper(token_string):
+    """gets the set of results that matches all the tokens in the token_string
+    
+    Args:
+        token_string: list of tokens as a string separated by ^
+    
+    Returns:
+        set of values that matches all the tokens in the token_string
+    """
+    tokens = token_string.split("^")
+    last_token_result = set(current_app.config['STAT_VAR_SEARCH_INDEX'].get(
+        tokens[-1], {}))
+    if len(tokens) == 1:
+        return last_token_result
+    prev_tokens = tokens[:(len(tokens) - 1)]
+    prev_tokens_string = "^".join(prev_tokens)
+    prev_intersection = search_helper(prev_tokens_string)
```

Comment:
I thought to make it a recursive function so that when there's multiple tokens in the query, we can make use of the caching and not have to recalculate the intersection of the previous tokens because those previous tokens should've already been seen

Reply:
okay, then may be the "cached" function should take an individual token? Cache is usually used when there is RPC involved. Since here all data is in memory and the operation is not heavy (set intersection), so caching might not be a big concern.